redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt

```diff
@@ -0,0 +1,2041 @@
+          continue
+        }
+        childTop += child.margin.top.toFloat()
+        childBottom -= child.margin.bottom.toFloat()
+        if (isRtl) {
+          if (fromBottomToTop) {
+            layoutSingleChildVertical(
+              node = child,
+              flexLine = flexLine,
+              isRtl = true,
+              left = childRight - child.measuredWidth,
+              top = childBottom.roundToInt() - child.measuredHeight,
+              right = childRight,
+              bottom = childBottom.roundToInt(),
+            )
+          } else {
+            layoutSingleChildVertical(
+              node = child,
+              flexLine = flexLine,
+              isRtl = true,
+              left = childRight - child.measuredWidth,
+              top = childTop.roundToInt(),
+              right = childRight,
+              bottom = childTop.roundToInt() + child.measuredHeight,
+            )
+          }
+        } else {
+          if (fromBottomToTop) {
+            layoutSingleChildVertical(
+              node = child,
+              flexLine = flexLine,
+              isRtl = false,
+              left = childLeft,
+              top = childBottom.roundToInt() - child.measuredHeight,
+              right = childLeft + child.measuredWidth,
+              bottom = childBottom.roundToInt(),
+            )
+          } else {
+            layoutSingleChildVertical(
+              node = child,
+              flexLine = flexLine,
+              isRtl = false,
+              left = childLeft,
+              top = childTop.roundToInt(),
+              right = childLeft + child.measuredWidth,
+              bottom = childTop.roundToInt() + child.measuredHeight,
+            )
+          }
+        }
+        childTop += child.measuredHeight + spaceBetweenItem + child.margin.bottom
+        childBottom -= child.measuredHeight + spaceBetweenItem + child.margin.top
+      }
+      childLeft += flexLine.crossSize
+      childRight -= flexLine.crossSize
+      i++
+    }
+  }
+
+  /**
+   * The largest main size of all flex lines including decorator lengths.
+   */
+  private fun getLargestMainSize(): Int {
+    return if (flexLines.isEmpty()) Int.MIN_VALUE else flexLines.maxOf { it.mainSize }
+  }
+
+  /**
+   * The sum of the cross sizes of all flex lines including decorator lengths.
+   */
+  private fun getSumOfCrossSize(): Int {
+    return flexLines.sumOf { it.crossSize }
+  }
+
+  /**
+   * Returns a View, which is reordered by taking [Node.order] parameters into account.
+   *
+   * @param index the index of the view
+   * @return the reordered view, which [Node.order] is taken into account.
+   * If the index is negative or out of bounds of the number of contained views, returns `null`.
+   */
+  private fun getReorderedChildAt(index: Int): Node? {
+    if (indexToReorderedIndex == null) {
+      val sorted = nodes.withIndex().sortedWith(compareBy({ -it.value.order }, { it.index }))
```

Comment:
This will likely be another needless performance bomb . . . . accessing each child sorts all the nodes? Yuck